File: goscripts/source/player.go
package game

import ()

type Player struct {
	Name     string
	Position Position
	Friends  Players
	Room     *Room
	id       string
	//Bag    Items
}

type Players map[string]*Player

type PlayerMovement struct {
	PlayerId string
	OldPos   Position
	NewPos   Position
	Up       bool
	Down     bool
	Right    bool
	Left     bool
}

func NewPlayer(playerName string, gameId string) *Player {
	id := newId()
	for Games[id] != nil {
		id = newId()
	}
	room := Games[gameId].StartRoom
	player := &Player{
		Name: playerName,
		id:   id,
		Room: room,
		Position: Position{
			X: 0,
			Y: 0,
		},
	}
	Games[gameId].players[id] = player
	return player
}

func (p *Player) GetId() string {
	return p.id
}

File: game/player.js
function player(player_info){
  speed = 1;
  position    = {x: 0, y: 0};
  constraints = {x: 30, y: 30};

  this.name = player_info.name;

  function move(movements) {
    if (movements.left || movements.right && !(movements.left && movements.right)) {
      if (movements.left && position.x > 0) {
        position.x -= speed;
      } else if (position.x <  constraints.x) {
        position.x += speed;
      }
    }
    if (movements.up || movements.down && !(movements.up && movements.down)) {
      if(movements.up && position.y > 0) {
        position.y -= speed;
      } else if (position.y < constraints.y) {
        position.y += speed;
      }
    }
  }

  return {
    handleRequest: function(data, io){
      if (data.movements) {
        move(data.movements)
      }
    },
    info: {
      position: position
    }
  }
}

module.exports = player;
